pkg/machines/components/storagePools/storagePoolOverviewTab.jsx

```diff
@@ -40,9 +40,19 @@ export const StoragePoolOverviewTab = ({ storagePool, actionError, actionErrorDe
                 </div>
             </React.Fragment> }
 
+            { storagePool.source && storagePool.source.device && <React.Fragment>
+                <label className='control-label' htmlFor={`${idPrefix}-source-path`}> {_("Source Path")} </label>
+                <div id={`${idPrefix}-source-path`}> {storagePool.source.device.path} </div>
+            </React.Fragment> }
+
+            { storagePool.source && storagePool.source.dir && <React.Fragment>
+                <label className='control-label' htmlFor={`${idPrefix}-source-path`}> {_("Source Path")} </label>
+                <div id={`${idPrefix}-source-path`}> {storagePool.source.dir.path} </div>
+            </React.Fragment> }
```

Comment:
Are these guaranteed to be mutually exclusive? I. e. device is always remote, dir is always local?

Reply:
The difference is not about remote or local source.  From libvirt documentation:

> **device**
> Provides the source for pools backed by physical devices (pool types fs, logical, disk, iscsi, iscsi-direct, zfs, vstorage).
> ...
> 
> **dir**
> Provides the source for pools backed by directories (pool types dir, netfs, gluster), or optionally to select a subdirectory within a pool that resembles a filesystem (pool type gluster).
> ...

There is also third source element, `<adapter>` which is used for `scsi` storage pools.

However, I've just noticed that there is a mistake in our documentation, storage pool `dir` has empty `<source>` element.

Based on the documentation and code inspection `device, dir and adapter` are mutually exclusive.

Reply:
@martinpitt Do you want that I will re-write this code to show off that these attributes are mutually exclusive. In my opinion, adding this knowledge in this code, doesn't offer some advantage.

Reply:
@KKoukiou : it's fine, I just want to understand it and make sure it works as intended.